examples/piggybank/src/Root.tsx

```diff
@@ -198,23 +224,31 @@ function PiggyBank() {
 export default function Root() {
     const [account, setAccount] = useState<string>();
     const [isConnected, setIsConnected] = useState<boolean>(false);
+    const [provider, setProvider] = useState<WalletApi>();
 
     useEffect(() => {
-        apiReady
-            // Connect to the wallet.
-            .then(() => window.concordium?.connect())
-            .then((acc) => {
-                // Connection accepted, set the application state parameters.
-                setAccount(acc);
-                setIsConnected(true);
-
-                // Listen for relevent events from the wallet.
-                window.concordium?.addChangeAccountListener(setAccount);
-            })
-            // Connection rejected.
-            .catch(() => setIsConnected(false));
+        detectConcordiumProvider().then(setProvider);
     }, []);
 
+    useEffect(() => {
+        if (provider) {
+            provider
+                .connect()
+                .then((acc) => {
+                    // Connection accepted, set the application state parameters.
+                    setAccount(acc);
+                    setIsConnected(true);
+
+                    // Listen for relevant events from the wallet.
+                    provider.addChangeAccountListener(setAccount);
+                })
+                .catch(() => setIsConnected(false));
+        } else {
+            // Not yet connected to the wallet.
+            setIsConnected(false);
+        }
+    }, [provider]);
```

Comment:
All this (including the provider state hook) seems a bit unnecessary? Why not just chain the connect and the subsequent stuff directly on `detectConcordiumProvider`?

Reply:
I thought that I had a good reason for doing it, but when looking at it now I can't see it. So yes, I will collapse all this.